docs/source/tutorial-pyfunc.rst

```diff
@@ -0,0 +1,368 @@
+
+The ``IrisClassifer`` class was defined in the previous section. It extends the :py:class:`mlflow.pyfunc.PythonModel` class. It defines the two necessary functions:
+
+The customized ``load_context`` implementation
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The ``load_context`` method is called when the model is loaded. It takes a single argument, ``context``, which is an instance of the :py:class:`mlflow.pyfunc.PythonModelContext` class. The most commonly used attribute of this class is ``artifacts``, which is a dictionary recreating the one passed to the ``mlflow.pyfunc.log_model`` method. 
+
+In this case, the ``artifacts`` dictionary contains a single key, ``"classifier"``, which has the value of ``"iris_classifier.joblib"``. The ``load_context`` method uses the ``joblib`` library to load the model from the file: ``self.classifier = jl.load(context.artifacts["classifier"])``. The classifier is then stored in the ``self.classifier`` attribute of the ``IrisClassifier`` instance.
+
+The customized ``predict`` implementation
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The other method necessary in a customized ``PythonModel`` is ``predict()``. This function takes two arguments:
+
+- ``context`` : The same ``PythonModelContext`` instance that was passed to ``load_context``.
+- ``model_input`` : A pyfunc-compatible input to the model. In this case, it is a Pandas DataFrame.
+
+In this case, the ``model_input`` is not validated or transformed, but is immediately passed to the ``predict`` method of the ``IrisClassifier`` model assigned to ``self.classifier`` in the ``load_context`` method. The results are then returned.
+
+.. note:: 
+    Notice that while the custom model's ``predict`` method takes both a ``context`` and a ``model_input`` parameter, the call to ``predict`` in **infer.py** only passes the ``model_input`` argument. The ``context`` parameter is set automatically by the base ``PythonModel`` implementation. 
+
+Querying the customized model
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Returning to **infer.py**, after the model is loaded, the ``load_and_use`` function creates a Pandas DataFrame containing three rows of data. The ``predict`` method of the loaded model is then called with the DataFrame as an argument. The results are printed to the console.
+
+Use the CLI to run and query the customized model
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Rather than the Python code in **infer.py**, you can instead use the MLflow CLI to run the customized model. Run the following command, replacing ``<RUNID>`` with the appropriate value:
+
+.. code-block:: bash
+
+    mlflow models serve -m runs:/<RUNID>/artifacts --port 5002
+
+Note the use of the ``--port`` argument to avoid the model serving conflicting with the MLflow Tracking Server (likely running on port 5000). 
+
+To query the model from the CLI, run the following command:
+
```

Comment:
Can we add an example of querying this endpoint using the `requests` library, as that will be a far more common pattern for our users.